File: compare_resources.php
<?php
/**
 * Resources Directory Comparison Script
 * Compares resources between current and old Laravel projects
 */

class ResourceComparator {
    private $currentProject = 'c:\\Projects\\vaagaacademy';
    private $oldProject = 'C:\\Users\\malik\\Downloads\\_public_html_vaaga_mock';
    
    public function run() {
        echo "=== RESOURCES DIRECTORY COMPARISON ===\n\n";
        
        // Check if directories exist
        echo "Checking project paths...\n";
        echo "Current project: " . $this->currentProject . " - " . (is_dir($this->currentProject) ? "EXISTS" : "NOT FOUND") . "\n";
        echo "Old project: " . $this->oldProject . " - " . (is_dir($this->oldProject) ? "EXISTS" : "NOT FOUND") . "\n\n";
        
        if (!is_dir($this->currentProject)) {
            echo "ERROR: Current project directory not found!\n";
            return;
        }
        
        if (!is_dir($this->oldProject)) {
            echo "ERROR: Old project directory not found!\n";
            return;
        }
        
        // 1. Compare Blade Views
        echo "========================================\n";
        echo "1. BLADE VIEW FILES COMPARISON\n";
        echo "========================================\n";
        $this->compareBladeViews();
        
        // 2. Compare JS Assets
        echo "\n========================================\n";
        echo "2. JAVASCRIPT ASSETS COMPARISON\n";
        echo "========================================\n";
        $this->compareJSAssets();
        
        // 3. Compare CSS/SASS Assets
        echo "\n========================================\n";
        echo "3. CSS/SASS ASSETS COMPARISON\n";
        echo "========================================\n";
        $this->compareCSSAssets();
        
        // 4. Compare Translation Files
        echo "\n========================================\n";
        echo "4. TRANSLATION FILES COMPARISON\n";
        echo "========================================\n";
        $this->compareTranslations();
        
        // 5. Overall Directory Structure
        echo "\n========================================\n";
        echo "5. RESOURCES DIRECTORY STRUCTURE\n";
        echo "========================================\n";
        $this->compareDirectoryStructure();
    }
    
    private function compareBladeViews() {
        $currentViews = $this->getBladeFiles($this->currentProject . '\\resources\\views');
        $oldViews = $this->getBladeFiles($this->oldProject . '\\resources\\views');
        
        echo "\n--- Current Project Blade Files ---\n";
        echo "Total: " . count($currentViews) . " files\n";
        
        echo "\n--- Old Project Blade Files ---\n";
        echo "Total: " . count($oldViews) . " files\n";
        
        // Find missing files
        $missingInCurrent = array_diff($oldViews, $currentViews);
        $missingInOld = array_diff($currentViews, $oldViews);
        
        echo "\n--- MISSING IN CURRENT PROJECT ---\n";
        if (empty($missingInCurrent)) {
            echo "No blade files missing in current project.\n";
        } else {
            echo "Count: " . count($missingInCurrent) . " files\n";
            foreach ($missingInCurrent as $file) {
                echo "  - " . $file . "\n";
            }
        }
        
        echo "\n--- NEW IN CURRENT PROJECT ---\n";
        if (empty($missingInOld)) {
            echo "No new blade files in current project.\n";
        } else {
            echo "Count: " . count($missingInOld) . " files\n";
            foreach ($missingInOld as $file) {
                echo "  - " . $file . "\n";
            }
        }
        
        return $missingInCurrent;
    }
    
    private function compareJSAssets() {
        $currentJS = $this->getFilesByExtension($this->currentProject . '\\resources', ['js']);
        $oldJS = $this->getFilesByExtension($this->oldProject . '\\resources', ['js']);
        
        echo "\n--- Current Project JS Files ---\n";
        echo "Total: " . count($currentJS) . " files\n";
        
        echo "\n--- Old Project JS Files ---\n";
        echo "Total: " . count($oldJS) . " files\n";
        
        // Find missing files
        $missingInCurrent = array_diff($oldJS, $currentJS);
        $missingInOld = array_diff($currentJS, $oldJS);
        
        echo "\n--- MISSING IN CURRENT PROJECT ---\n";
        if (empty($missingInCurrent)) {
            echo "No JS files missing in current project.\n";
        } else {
            echo "Count: " . count($missingInCurrent) . " files\n";
            foreach ($missingInCurrent as $file) {
                echo "  - " . $file . "\n";
            }
        }
        
        echo "\n--- NEW IN CURRENT PROJECT ---\n";
        if (empty($missingInOld)) {
            echo "No new JS files in current project.\n";
        } else {
            echo "Count: " . count($missingInOld) . " files\n";
            foreach ($missingInOld as $file) {
                echo "  - " . $file . "\n";
            }
        }
    }
    
    private function compareCSSAssets() {
        $currentCSS = $this->getFilesByExtension($this->currentProject . '\\resources', ['css', 'scss', 'sass']);
        $oldCSS = $this->getFilesByExtension($this->oldProject . '\\resources', ['css', 'scss', 'sass']);
        
        echo "\n--- Current Project CSS/SCSS/SASS Files ---\n";
        echo "Total: " . count($currentCSS) . " files\n";
        
        echo "\n--- Old Project CSS/SCSS/SASS Files ---\n";
        echo "Total: " . count($oldCSS) . " files\n";
        
        // Find missing files
        $missingInCurrent = array_diff($oldCSS, $currentCSS);
        $missingInOld = array_diff($currentCSS, $oldCSS);
        
        echo "\n--- MISSING IN CURRENT PROJECT ---\n";
        if (empty($missingInCurrent)) {
            echo "No CSS/SCSS/SASS files missing in current project.\n";
        } else {
            echo "Count: " . count($missingInCurrent) . " files\n";
            foreach ($missingInCurrent as $file) {
                echo "  - " . $file . "\n";
            }
        }
        
        echo "\n--- NEW IN CURRENT PROJECT ---\n";
        if (empty($missingInOld)) {
            echo "No new CSS/SCSS/SASS files in current project.\n";
        } else {
            echo "Count: " . count($missingInOld) . " files\n";
            foreach ($missingInOld as $file) {
                echo "  - " . $file . "\n";
            }
        }
    }
    
    private function compareTranslations() {
        $currentLang = $this->getAllFiles($this->currentProject . '\\resources\\lang');
        $oldLang = $this->getAllFiles($this->oldProject . '\\resources\\lang');
        
        echo "\n--- Current Project Translation Files ---\n";
        echo "Total: " . count($currentLang) . " files\n";
        foreach ($currentLang as $file) {
            echo "  - " . $file . "\n";
        }
        
        echo "\n--- Old Project Translation Files ---\n";
        echo "Total: " . count($oldLang) . " files\n";
        foreach ($oldLang as $file) {
            echo "  - " . $file . "\n";
        }
        
        // Find missing files
        $missingInCurrent = array_diff($oldLang, $currentLang);
        $missingInOld = array_diff($currentLang, $oldLang);
        
        echo "\n--- MISSING IN CURRENT PROJECT ---\n";
        if (empty($missingInCurrent)) {
            echo "No translation files missing in current project.\n";
        } else {
            echo "Count: " . count($missingInCurrent) . " files\n";
            foreach ($missingInCurrent as $file) {
                echo "  - " . $file . "\n";
            }
        }
        
        echo "\n--- NEW IN CURRENT PROJECT ---\n";
        if (empty($missingInOld)) {
            echo "No new translation files in current project.\n";
        } else {
            echo "Count: " . count($missingInOld) . " files\n";
            foreach ($missingInOld as $file) {
                echo "  - " . $file . "\n";
            }
        }
    }
    
    private function compareDirectoryStructure() {
        $currentStructure = $this->getDirectoryStructure($this->currentProject . '\\resources');
        $oldStructure = $this->getDirectoryStructure($this->oldProject . '\\resources');
        
        echo "\n--- Current Project Structure ---\n";
        foreach ($currentStructure as $dir => $count) {
            echo "  $dir: $count files\n";
        }
        
        echo "\n--- Old Project Structure ---\n";
        foreach ($oldStructure as $dir => $count) {
            echo "  $dir: $count files\n";
        }
        
        // Directories only in old
        $onlyInOld = array_diff_key($oldStructure, $currentStructure);
        echo "\n--- Directories only in Old Project ---\n";
        if (empty($onlyInOld)) {
            echo "None\n";
        } else {
            foreach ($onlyInOld as $dir => $count) {
                echo "  $dir: $count files\n";
            }
        }
        
        // Directories only in current
        $onlyInCurrent = array_diff_key($currentStructure, $oldStructure);
        echo "\n--- Directories only in Current Project ---\n";
        if (empty($onlyInCurrent)) {
            echo "None\n";
        } else {
            foreach ($onlyInCurrent as $dir => $count) {
                echo "  $dir: $count files\n";
            }
        }
    }
    
    private function getBladeFiles($directory) {
        return $this->getFilesByExtension($directory, ['blade.php']);
    }
    
    private function getFilesByExtension($directory, $extensions) {
        $files = [];
        
        if (!is_dir($directory)) {
            return $files;
        }
        
        $iterator = new RecursiveIteratorIterator(
            new RecursiveDirectoryIterator($directory, RecursiveDirectoryIterator::SKIP_DOTS)
        );
        
        foreach ($iterator as $file) {
            if ($file->isFile()) {
                $filename = $file->getFilename();
                foreach ($extensions as $ext) {
                    if (substr($filename, -strlen('.'.$ext)) === '.'.$ext) {
                        // Store relative path from resources directory
                        $relativePath = str_replace($this->currentProject . '\\resources\\', '', $file->getPathname());
                        $relativePath = str_replace($this->oldProject . '\\resources\\', '', $relativePath);
                        $files[] = $relativePath;
                        break;
                    }
                }
            }
        }
        
        sort($files);
        return $files;
    }
    
    private function getAllFiles($directory) {
        $files = [];
        
        if (!is_dir($directory)) {
            return $files;
        }
        
        $iterator = new RecursiveIteratorIterator(
            new RecursiveDirectoryIterator($directory, RecursiveDirectoryIterator::SKIP_DOTS)
        );
        
        foreach ($iterator as $file) {
            if ($file->isFile()) {
                $relativePath = str_replace($directory, '', $file->getPathname());
                $relativePath = ltrim($relativePath, '\\/');
                $files[] = $relativePath;
            }
        }
        
        sort($files);
        return $files;
    }
    
    private function getDirectoryStructure($directory) {
        $structure = [];
        
        if (!is_dir($directory)) {
            return $structure;
        }
        
        $directories = new RecursiveIteratorIterator(
            new RecursiveDirectoryIterator($directory, RecursiveDirectoryIterator::SKIP_DOTS),
            RecursiveIteratorIterator::SELF_FIRST
        );
        
        foreach ($directories as $item) {
            if ($item->isDir()) {
                $relativePath = str_replace($directory . '\\', '', $item->getPathname());
                $fileCount = count(glob($item->getPathname() . '\\*'));
                $structure[$relativePath] = $fileCount;
            }
        }
        
        return $structure;
    }
}

// Run the comparison
$comparator = new ResourceComparator();
$comparator->run();

File: _archive/compare_resources.py
#!/usr/bin/env python3
"""
Resources Directory Comparison Script
Compares resources between current and old Laravel projects
"""

import os
from pathlib import Path
from collections import defaultdict

def get_all_files(directory, extensions=None):
    """Get all files in directory with optional extension filtering"""
    files = []
    if not os.path.exists(directory):
        return files
    
    for root, dirs, filenames in os.walk(directory):
        for filename in filenames:
            if extensions:
                if any(filename.endswith(ext) for ext in extensions):
                    full_path = os.path.join(root, filename)
                    rel_path = os.path.relpath(full_path, directory)
                    files.append(rel_path.replace('\\', '/'))
            else:
                full_path = os.path.join(root, filename)
                rel_path = os.path.relpath(full_path, directory)
                files.append(rel_path.replace('\\', '/'))
    
    return sorted(files)

def get_directory_structure(directory):
    """Get directory structure with file counts"""
    structure = {}
    if not os.path.exists(directory):
        return structure
    
    for root, dirs, files in os.walk(directory):
        rel_path = os.path.relpath(root, directory)
        if rel_path == '.':
            rel_path = ''
        structure[rel_path.replace('\\', '/')] = len(files)
    
    return structure

def main():
    current_project = r'c:\Projects\vaagaacademy'
    old_project = r'C:\Users\malik\Downloads\_public_html_vaaga_mock'
    
    print("=" * 60)
    print("RESOURCES DIRECTORY COMPARISON")
    print("=" * 60)
    
    # Check if directories exist
    print("\nChecking project paths...")
    print(f"Current project: {current_project} - {'EXISTS' if os.path.exists(current_project) else 'NOT FOUND'}")
    print(f"Old project: {old_project} - {'EXISTS' if os.path.exists(old_project) else 'NOT FOUND'}")
    
    if not os.path.exists(current_project):
        print("ERROR: Current project directory not found!")
        return
    
    if not os.path.exists(old_project):
        print("ERROR: Old project directory not found!")
        return
    
    # 1. Blade View Files Comparison
    print("\n" + "=" * 60)
    print("1. BLADE VIEW FILES COMPARISON")
    print("=" * 60)
    
    current_blade = get_all_files(os.path.join(current_project, 'resources', 'views'), ['.blade.php'])
    old_blade = get_all_files(os.path.join(old_project, 'resources', 'views'), ['.blade.php'])
    
    print(f"\nCurrent Project: {len(current_blade)} blade files")
    print(f"Old Project: {len(old_blade)} blade files")
    
    missing_blade = sorted(set(old_blade) - set(current_blade))
    new_blade = sorted(set(current_blade) - set(old_blade))
    
    print(f"\n--- MISSING IN CURRENT PROJECT ({len(missing_blade)} files) ---")
    for f in missing_blade:
        print(f"  - {f}")
    
    print(f"\n--- NEW IN CURRENT PROJECT ({len(new_blade)} files) ---")
    for f in new_blade[:20]:  # Limit output
        print(f"  - {f}")
    if len(new_blade) > 20:
        print(f"  ... and {len(new_blade) - 20} more files")
    
    # 2. JavaScript Assets Comparison
    print("\n" + "=" * 60)
    print("2. JAVASCRIPT ASSETS COMPARISON")
    print("=" * 60)
    
    current_js = get_all_files(os.path.join(current_project, 'resources'), ['.js'])
    old_js = get_all_files(os.path.join(old_project, 'resources'), ['.js'])
    
    print(f"\nCurrent Project: {len(current_js)} JS files")
    print(f"Old Project: {len(old_js)} JS files")
    
    missing_js = sorted(set(old_js) - set(current_js))
    new_js = sorted(set(current_js) - set(old_js))
    
    print(f"\n--- MISSING IN CURRENT PROJECT ({len(missing_js)} files) ---")
    for f in missing_js:
        print(f"  - {f}")
    
    print(f"\n--- NEW IN CURRENT PROJECT ({len(new_js)} files) ---")
    for f in new_js:
        print(f"  - {f}")
    
    # 3. CSS/SASS Assets Comparison
    print("\n" + "=" * 60)
    print("3. CSS/SASS ASSETS COMPARISON")
    print("=" * 60)
    
    current_css = get_all_files(os.path.join(current_project, 'resources'), ['.css', '.scss', '.sass'])
    old_css = get_all_files(os.path.join(old_project, 'resources'), ['.css', '.scss', '.sass'])
    
    print(f"\nCurrent Project: {len(current_css)} CSS/SCSS/SASS files")
    print(f"Old Project: {len(old_css)} CSS/SCSS/SASS files")
    
    missing_css = sorted(set(old_css) - set(current_css))
    new_css = sorted(set(current_css) - set(old_css))
    
    print(f"\n--- MISSING IN CURRENT PROJECT ({len(missing_css)} files) ---")
    for f in missing_css:
        print(f"  - {f}")
    
    print(f"\n--- NEW IN CURRENT PROJECT ({len(new_css)} files) ---")
    for f in new_css:
        print(f"  - {f}")
    
    # 4. Translation Files Comparison
    print("\n" + "=" * 60)
    print("4. TRANSLATION FILES COMPARISON")
    print("=" * 60)
    
    current_lang = get_all_files(os.path.join(current_project, 'resources', 'lang'))
    old_lang = get_all_files(os.path.join(old_project, 'resources', 'lang'))
    
    print(f"\nCurrent Project: {len(current_lang)} translation files")
    print(f"Old Project: {len(old_lang)} translation files")
    
    missing_lang = sorted(set(old_lang) - set(current_lang))
    new_lang = sorted(set(current_lang) - set(old_lang))
    
    print(f"\n--- MISSING IN CURRENT PROJECT ({len(missing_lang)} files) ---")
    for f in missing_lang[:30]:  # Limit output
        print(f"  - {f}")
    if len(missing_lang) > 30:
        print(f"  ... and {len(missing_lang) - 30} more files")
    
    print(f"\n--- NEW IN CURRENT PROJECT ({len(new_lang)} files) ---")
    for f in new_lang[:10]:  # Limit output
        print(f"  - {f}")
    if len(new_lang) > 10:
        print(f"  ... and {len(new_lang) - 10} more files")
    
    # 5. Directory Structure Comparison
    print("\n" + "=" * 60)
    print("5. RESOURCES DIRECTORY STRUCTURE")
    print("=" * 60)
    
    current_structure = get_directory_structure(os.path.join(current_project, 'resources'))
    old_structure = get_directory_structure(os.path.join(old_project, 'resources'))
    
    print("\n--- Current Project Structure ---")
    for dir_path in sorted(current_structure.keys())[:30]:
        if dir_path:
            print(f"  {dir_path}: {current_structure[dir_path]} files")
    
    print("\n--- Old Project Structure ---")
    for dir_path in sorted(old_structure.keys())[:30]:
        if dir_path:
            print(f"  {dir_path}: {old_structure[dir_path]} files")
    
    # Summary
    print("\n" + "=" * 60)
    print("SUMMARY")
    print("=" * 60)
    print(f"\nMissing Blade Views: {len(missing_blade)}")
    print(f"Missing JS Assets: {len(missing_js)}")
    print(f"Missing CSS/SASS Assets: {len(missing_css)}")
    print(f"Missing Translation Files: {len(missing_lang)}")
    
    # Save detailed results to file
    output_file = os.path.join(current_project, 'resources_comparison_report.txt')
    with open(output_file, 'w', encoding='utf-8') as f:
        f.write("=" * 80 + "\n")
        f.write("RESOURCES DIRECTORY COMPARISON REPORT\n")
        f.write("=" * 80 + "\n\n")
        
        f.write("MISSING BLADE VIEW FILES:\n")
        f.write("-" * 80 + "\n")
        for item in missing_blade:
            f.write(f"  {item}\n")
        f.write(f"\nTotal: {len(missing_blade)} files\n\n")
        
        f.write("MISSING JS ASSETS:\n")
        f.write("-" * 80 + "\n")
        for item in missing_js:
            f.write(f"  {item}\n")
        f.write(f"\nTotal: {len(missing_js)} files\n\n")
        
        f.write("MISSING CSS/SASS ASSETS:\n")
        f.write("-" * 80 + "\n")
        for item in missing_css:
            f.write(f"  {item}\n")
        f.write(f"\nTotal: {len(missing_css)} files\n\n")
        
        f.write("MISSING TRANSLATION FILES:\n")
        f.write("-" * 80 + "\n")
        for item in missing_lang:
            f.write(f"  {item}\n")
        f.write(f"\nTotal: {len(missing_lang)} files\n\n")
        
        f.write("NEW IN CURRENT PROJECT:\n")
        f.write("-" * 80 + "\n")
        f.write(f"Blade Views: {len(new_blade)} files\n")
        f.write(f"JS Assets: {len(new_js)} files\n")
        f.write(f"CSS/SASS Assets: {len(new_css)} files\n")
        f.write(f"Translation Files: {len(new_lang)} files\n")
    
    print(f"\nDetailed report saved to: {output_file}")

if __name__ == '__main__':
    main()
